### PR TITLE
Feat/searchbar

### PR DESCRIPTION
**Pull request by marcelherr**
add Searchbar component that filters books by author and title

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -29,22 +28,28 @@ export default function App() {
...
     useEffect(() => {
         fetchBooks()
     }, []);
 
+    const [searchInput, setSearchInput] = useState("")
+
+    const filteredBooks: Book[] = data
+        .filter((book) => book.title.toLowerCase().includes(searchInput.toLowerCase()) ||
+            book.author.toLowerCase().includes(searchInput.toLowerCase()));
+
```
Comment (rinaehyun) [inline frontend/src/App.tsx:42]: To make sure if the data is available when the app is running, I suggest the following code.

```typescript
const filteredBooks: Book[] = data.length > 0 ?
        data.filter((book) => book.title?.toLowerCase().includes(searchInput.toLowerCase()) ||
            book.author?.toLowerCase().includes(searchInput.toLowerCase()))
        : [];
``` `

Comment (esgoet) [reply frontend/src/App.tsx:42]: but wouldn't it be ok if data is just an empty array? then there is just nothing to filter

File: frontend/src/pages/BookGalleryPage/components/searchbar/Searchbar.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import './Searchbar.css'
+
+export default function Searchbar({setSearchInput}: any) {
```
Comment (esgoet) [inline frontend/src/pages/BookGalleryPage/components/searchbar/Searchbar.tsx:3]:
```suggestion
export default function SearchBar({setSearchInput}: any) {
```

Comment (esgoet) [reply frontend/src/pages/BookGalleryPage/components/searchbar/Searchbar.tsx:3]: would change it to SearchBar, also for file names

File: frontend/src/pages/BookGalleryPage/components/searchbar/Searchbar.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    return (
+        <>
+            <label className={"search-label"} htmlFor={"searchbar"}>
```
Comment (esgoet) [inline frontend/src/pages/BookGalleryPage/components/searchbar/Searchbar.tsx:6]: so far in the project, the labels don't go around the input (that's why we have the htmlFor), instead I would just a div for styling that includes the label and the input

**Review by SimonSta95 — APPROVED** [review]
lgtm

**Pull request merged**